lib/absinthe/phase/schema.ex

```diff
@@ -161,6 +161,9 @@ defmodule Absinthe.Phase.Schema do
 
   defp set_schema_node(%Blueprint.Input.Field{} = node, parent, schema, adapter) do
     case node.name do
+      name = "__inputname" ->
+        %{node | schema_node: find_schema_field(parent.schema_node, name, node, schema, adapter)}
+
```

Comment:
necessary

Reply:
could you modify the `__` case to specifically allow this through

```
"__" <> introspection when introspection not in ["inputname"] ->
```

then we aren't duplicating the same code